src/daemon/daemon.cpp

```diff
@@ -178,8 +188,10 @@ auto make_cloud_init_image(const std::string& name, const QDir& instance_dir, YA
     mp::CloudInitIso iso;
     iso.add_file("meta-data", mpu::emit_cloud_config(meta_data_config));
     iso.add_file("vendor-data", mpu::emit_cloud_config(vendor_data_config));
-    iso.add_file("network-config", mpu::emit_cloud_config(network_data_config));
     iso.add_file("user-data", mpu::emit_cloud_config(user_data_config));
+    if (!network_data_config.IsNull())
+        iso.add_file("network-config", mpu::emit_cloud_config(network_data_config));
```

Comment:
Care to test this one too? :slightly_smiling_face:

Reply:
This line seems to be tested now: https://codecov.io/gh/canonical/multipass/pull/1913/src/src/daemon/daemon.cpp?before=src/daemon/daemon.cpp

In fact, I am surprised by it not being covered at the moment you made the comment (I checked that it was indeed uncovered at that point).